tests/integration/test_charm.py

```diff
@@ -31,7 +36,7 @@ async def test_build_and_deploy(ops_test: OpsTest):
         charm_under_test, resources=resources, application_name=APP_NAME, trust=True
     )
     await ops_test.model.wait_for_idle(
-        apps=[APP_NAME], status="active", raise_on_blocked=True, timeout=60 * 10
+        apps=[APP_NAME], status="active", raise_on_blocked=False, timeout=60 * 10
```

Comment:
Is there a condition where the charm blocks and then unblocks itself that we don't want to catch during testing?

Reply:
That happens if the training-operator was previously deployed in a controller and then re-created. Unit gets blocked with message `(leader-elected) Creating/patching resources failed with code 409` because the `kubeflow-training-admin` ClusterRole wasn't deleted along with the previous deployment. Lightkube raises this status due to leftovers and that causes the `install` hook to put the unit into `BlockedStatus`. But then the `config-changed` hook runs and patches the resources which makes the unit active without manual intervention.
This exception was handled for crds, but not for auth resources to which the ClusterRole belongs - I added it and reverted to `raise_on_blocked=True` in the integration test.

Reply:
This sounds like we have a separate problem that not everything deployed by training operator gets cleaned up when juju removes the application - are there things we can do to fix that root cause?  Or is it a juju bug we can open?